efs/variables.tf

```diff
@@ -7,3 +7,8 @@ variable "access_point_path" {
 variable "policy" {
   default = "restrict_access_points"
 }
+
+variable "mount_target_security_groups" {
+  description = "Security groups which are allowed to access the mount target"
+  type        = list(list(string))
```

Comment:
Does this need to be a list of a list, rather than a flat list? It's just a bit of a weird type for the module interface to have.

Reply:
Because the security groups in the individual lambda/ecs files have a count on them to stop them being created without the correct filter on, the output is a list.
```
resource "aws_security_group" "allow_efs_lambda_download_files" {
  count       = local.count_download_files
....
```
```
output "download_files_lambda_sg_id" {
  value = aws_security_group.allow_efs_lambda_download_files.*.id
}
```
We can't choose a particular element, i.e. aws_security_group.allow_efs_lambda_download_files[0].id because when the count is 0, the list is empty and you get an error. So each output is a list.

```
[file-format-security-group]
[download-files-security-group]
[ecs-file-format-build-security-group]
```

These are added to the list for the efs module and are then flattened into a single list. 
`security_groups = flatten(var.mount_target_security_groups)`

Reply:
So is it possible to flatten them at the point where you call the module? It would make this interface simpler if it was `list(string)`.

Reply:
Yeah good point. I've called flatten in the root.tf file in the environments repo and changed it here to be a list(string)